tests/color_test.py

```diff
@@ -50,3 +51,20 @@ def test_use_color_tty_without_color_support():
 def test_use_color_raises_if_given_shenanigans():
     with pytest.raises(InvalidColorSetting):
         use_color('herpaderp')
+
+
+def test_no_color_env_unset():
+    with mock.patch.dict(os.environ):
```

Comment:
`clear=True` is what you want here I believe

Reply:
Wouldn't that remove all entries from `os.environ`? It would probably still work, though.

Reply:
yes, and yes

I guess if you want to be more direct you could use:

```python
with envcontext([('NO_COLOR', UNSET)]):
    ...
```

`envcotext` and `UNSET` come from `pre_commit.envcontext`

Reply:
Ok, I will change that because otherwise the coverage is reduced.